src/shared/json.c

```diff
@@ -1786,6 +1786,11 @@ int json_variant_format(JsonVariant *v, JsonFormatFlags flags, char **ret) {
         if (flags & JSON_FORMAT_OFF)
                 return -ENOEXEC;
 
+        if ((flags & JSON_FORMAT_REFUSE_SENSITIVE))
+                for (size_t i = 0; i < json_variant_elements(v); ++i)
+                        if (json_variant_is_sensitive(json_variant_by_index(v, i)))
+                                return -EPERM;
```

Comment:
this isn't a recursive check. it only checks the first level down, not even the top-level...

i.e. maybe add a new helper json_variant_contains_sensitive() which goes down the tree.

(note that JsonVariants are generally immutable for their content, but not their metadata, i.e. we allow the ref counter to change, and we allow the sensitive to be turned on (but not off) any time, and that means we unfortunately really have to do this check recursively...)

Reply:
Yeah misunderstood how this stuff was stored, fixed now